adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -1562,11 +1562,33 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
     stringency: ValidationStringency = ValidationStringency.STRICT)(implicit s: DummyImplicit): AlignmentRecordRDD = LoadIndexedBam.time {
 
     val path = new Path(pathName)
-    // todo: can this method handle SAM and CRAM, or just BAM?
+
+    // If pathName is a single file or *.bam, append .bai to find all bam indices.
+    // Otherwise, pathName is a directory and the entire path must be searched
+    // for indices.
+    val indexPath = if (pathName.endsWith(".bam")) {
+      new Path(pathName + ".bai")
+    } else {
+      path
+    }
+
+    // currently only supports BAM files, see https://github.com/bigdatagenomics/adam/issues/1833
     val bamFiles = getFsAndFiles(path).filter(p => p.toString.endsWith(".bam"))
 
+    val indexFiles = getFsAndFiles(indexPath).filter(p => p.toString.endsWith(".bai"))
```

Comment:
@fnothaft getFsAndFilesWith filter wasn't able to find .bai files unless just the directory was specified

Reply:
Yeah makes sense.